extensions-core/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesPeonLifecycle.java

```diff
@@ -349,6 +356,24 @@ protected void startWatchingLogs()
   }
 
   protected void saveLogs()
+  {
+    ExecutorService executor = Executors.newSingleThreadExecutor();
```

Comment:
Does it make sense to expose the timeouts here
https://github.com/fabric8io/kubernetes-client?tab=readme-ov-file#configuring-the-client
rather than doing a timeout via an exec service ?

Reply:
> Does it make sense to expose the timeouts here
> 
> https://github.com/fabric8io/kubernetes-client?tab=readme-ov-file#configuring-the-client
> 
> rather than doing a timeout via an exec service ?
> 
> 
> 
> 

https://github.com/fabric8io/kubernetes-client/issues/7163

From talking with @kgyrtkirk i was under the impression that the reason this log read can hang is because of the above bug. I don't think any existing timeouts can fix that. But we could also expose all the timeouts that fabric8 supplies as part of this PR?

Reply:
> this seems to create a new executor for every call? that looks a bit odd

new executor per task called once when task completes